src/drivers/navio_rgbled/navio_rgbled.cpp

```diff
@@ -0,0 +1,221 @@
+
+#include <px4_posix.h>
+#include <drivers/drv_rgbled.h>
+
+#include "navio_rgbled.h"
+
+#define GPIO_LED_CNF    (GPIO_CNF_OUTPUT)
+#define GPIO_LED_R      (GPIO_PIN4)
+#define GPIO_LED_G      (GPIO_PIN27)
+#define GPIO_LED_B      (GPIO_PIN6)
+
+using namespace DriverFramework;
+
+
+int RGBLED::start()
+{
+	int res;
+
+	res = DevObj::init();
+
+	if (res != 0) {
+		DF_LOG_ERR("error: could not init DevObj");
+		return res;
+	}
+
+	_gpio.start();
+
+	_gpio.configgpio(GPIO_LED_CNF | GPIO_LED_R);
+	_gpio.configgpio(GPIO_LED_CNF | GPIO_LED_G);
+	_gpio.configgpio(GPIO_LED_CNF | GPIO_LED_B);
+
+	return 0;
+}
+
+int RGBLED::stop()
+{
+	int res;
+
+	_gpio.stop();
+
+	res = DevObj::stop();
+
+	if (res < 0) {
+		DF_LOG_ERR("error: could not stop DevObj");
+		//this may not be an error for this device
+		return res;
+	}
+
+	return 0;
+}
+
+int RGBLED::devIOCTL(unsigned long request, unsigned long arg)
```

Comment:
One way would be to subscribe to the vehicle status and set the colours accordingly, however that's not possible from inside the DriverFramework. I'll think about this and get back to you.